### PR TITLE
Update ubuntu version in Dockerfile to 20.04

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Ubuntu 18.04 will reach EOL on April 30, 2023. We should avoid using it. As discovered in https://github.com/mlflow/mlflow/pull/7130#issuecomment-1286479759, Ubuntu < 20.04 might cause package installation issues when wheels for Ubuntu < 20.04 are unavailable.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/docker_utils.py [hunk 1]
```diff
@@ -31,7 +31,9 @@
     https://github.com/pyenv/pyenv.git /root/.pyenv
 ENV PYENV_ROOT="/root/.pyenv"
 ENV PATH="$PYENV_ROOT/bin:$PATH"
-RUN apt install -y python3.7
+RUN apt install -y software-properties-common \
+    && add-apt-repository -y ppa:deadsnakes/ppa \
+    && apt install -y python3.7
```
Comment (BenWilson2) [inline mlflow/models/docker_utils.py:36]: I think in 20.04 the distutils package is no longer considered "core python". 
`apt install -y python3.7-distutils` (and maybe `apt install -y python3.7-dev` are required now?)

Comment (harupy) [reply mlflow/models/docker_utils.py:36]: I'll try `apt install -y python3.7-distutils` :)

File: mlflow/models/docker_utils.py [hunk 2]
```diff
@@ -80,10 +82,10 @@ def _get_maven_proxy():
 
 _DOCKERFILE_TEMPLATE = """
 # Build an image that can serve mlflow models.
-FROM ubuntu:18.04
+FROM ubuntu:20.04
 
 RUN apt-get -y update
-RUN apt-get install -y --no-install-recommends \
+RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get install -y --no-install-recommends \
```
Comment (harupy) [inline mlflow/models/docker_utils.py:88]: Build hangs due to tzdata's prompt. `DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC` prevents it.

File: mlflow/models/docker_utils.py (hunk 1 above)
Comment (harupy) [reply mlflow/models/docker_utils.py]: Added :)

File: mlflow/models/docker_utils.py [hunk 3]
```diff
@@ -31,7 +31,9 @@
     https://github.com/pyenv/pyenv.git /root/.pyenv
 ENV PYENV_ROOT="/root/.pyenv"
 ENV PATH="$PYENV_ROOT/bin:$PATH"
-RUN apt install -y python3.7
+RUN apt install -y software-properties-common \
+    && add-apt-repository -y ppa:deadsnakes/ppa \
+    && apt install -y python3.7 python3.7-distutils
```
Comment (dbczumar) [inline mlflow/models/docker_utils.py:36]: If we're going to update MLflow to require Python 3.8, shall we update this as well? https://github.com/mlflow/mlflow/pull/7161/files#diff-848204af1ec28fbbde224caf1c529b15c8f432f2de419830ace7cf2eb33d6abd

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/7151/files#r1005131650 is addressed. Thanks @harupy !

File: mlflow/models/docker_utils.py (hunk 3 above)
Comment (harupy) [reply mlflow/models/docker_utils.py:36]: Got it, I'll update it.

Comment (harupy) [inline mlflow/models/docker_utils.py:36]:
```suggestion
RUN apt install -y python3.8
```

**Review by BenWilson2 — APPROVED** [review]
LGTM! Thank you for updating this :)

**Pull request merged**